rust/opendp/src/trans/variance/mod.rs

```diff
@@ -1,31 +1,32 @@
 use std::iter::Sum;
-use std::ops::{Div, Sub, Add};
+use std::ops::{Add, Div, Sub, Mul};
 
 use num::{Float, One, Zero};
 
 use crate::core::{Function, StabilityRelation, Transformation};
-use crate::dist::{SymmetricDistance, AbsoluteDistance, IntDistance};
+use crate::dist::{AbsoluteDistance, IntDistance, SymmetricDistance};
 use crate::dom::{AllDomain, BoundedDomain, SizedDomain, VectorDomain};
 use crate::error::Fallible;
-use crate::traits::{DistanceConstant, ExactIntCast, InfCast, CheckedMul, CheckNull};
-
+use crate::traits::{CheckNull, DistanceConstant, ExactIntCast, InfCast, InfSub, InfAdd, InfMul, NegInfAdd, NegInfSub};
 
 pub fn make_sized_bounded_variance<T>(
     size: usize, bounds: (T, T), ddof: usize
-) -> Fallible<Transformation<SizedDomain<VectorDomain<BoundedDomain<T>>>, AllDomain<T>, SymmetricDistance, AbsoluteDistance<T>>>
-    where T: DistanceConstant<IntDistance> + Float + One + Sub<Output=T> + Div<Output=T> + Sum<T> + for<'a> Sum<&'a T> + ExactIntCast<usize> + CheckedMul + CheckNull,
-          for<'a> &'a T: Sub<Output=T> + Add<&'a T, Output=T>,
-          IntDistance: InfCast<T> {
+) -> Fallible<Transformation<SizedDomain<VectorDomain<BoundedDomain<T>>>, AllDomain<T>, SymmetricDistance, AbsoluteDistance<T>>> where
+    T: DistanceConstant<IntDistance> + Float + One + Sub<Output=T> + Div<Output=T>
+    + Sum<T> + for<'a> Sum<&'a T> + ExactIntCast<usize>
+    + InfMul + InfSub + NegInfSub + InfAdd + NegInfAdd + CheckNull,
+    for<'a> &'a T: Sub<Output=T> + Add<&'a T, Output=T>,
+    IntDistance: InfCast<T> {
+
     let _size = T::exact_int_cast(size)?;
     let _ddof = T::exact_int_cast(ddof)?;
     let (lower, upper) = bounds.clone();
     let _1 = T::one();
-    let _2 = &_1 + &_1;
+    let _2 = T::exact_int_cast(2)?;
 
-    let range = (&upper - &lower) / _2.clone();
-    if range.clone().checked_mul(&range).is_none() {
-        return fallible!(MakeTransformation, "Detected potential for overflow when computing function.")
-    }
+    let range = upper.inf_sub(&lower)?;
+    // check for potential overflow
+    range.inf_div(&_2)?.inf_mul(&range.inf_div(&_2)?)?;
```

Comment:
A little explanation here would help.

Reply:
I loosened the check to be equal to the implicit check when computing the constant, but left it duplicated, because the constant calculation would become sensitive to ordering.